dataset/batch.py

```diff
@@ -568,19 +569,49 @@ def apply_transform_all(self, func, *args, src=None, dst=None, p=1., use_self=Fa
             dst[:] = tr_res
         return self
 
-    def _get_file_name(self, ix, src, ext):
-        if src is None:
-            if isinstance(self.index, FilesIndex):
-                src = self.index.get_fullpath(ix)
+    def _get_file_name(self, ix, src, write=False):
+        """ Get file name corresponding to current index.
+
+        Parameters
+        ----------
+        src : str or FilesIndex
+            if str, can be name of a file with extension in the indexed path (i.e self.index.dirs must be True)
+            if FilesIndex it must contain the same indices values as in the self.index.
+
+        write : bool
+            write = True is designed to be called from dump. Then src must be str and stand for a name of a folder
+            where data will be dumped in files named the same names as in current index.
+        """
+        if isinstance(self.index, FilesIndex):
+            if isinstance(src, str):
+                if write and not self.index.dirs:
+                    file_name = self.index.get_fullpath(ix).replace('\\', '/').split('/')[-1]
+                    if not os.path.exists(src):
+                        os.makedirs(src)
```

Comment:
Avoid side effects. This function should return a name without affecting a file system